examples/account_management/link_accounts.rb

```diff
@@ -0,0 +1,165 @@
+require 'google/ads/google_ads'
+
+def link_accounts(manager_customer_id, client_customer_id)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  # This example assumes that the same credentials will work for both accounts,
+  # but that may not be the case. If you need to use different credentials
+  # for each account, then you may either update the client configuration or
+  # instantiate two clients, one for each set of credentials. Always make sure
+  # to update the configuration before fetching any services you need to use.
+
+  # Extend an invitation while authenticating as the manager account.
+  client.configure do |config|
+    config.login_customer_id = manager_customer_id.to_i
+  end
+
+  client_link_service = client.service(:CustomerClientLink)
+
+  client_link = client.resource(:CustomerClientLink)
+  client_link.client_customer = client.wrapper.string(
+    client.path.customer(client_customer_id),
+  )
+  client_link.status = :PENDING
+
+  client_link_operation = client.operation(:CustomerClientLink)
+  client_link_operation['create'] = client_link
+
+  response = client_link_service.mutate_customer_client_link(
+    manager_customer_id,
+    client_link_operation,
+  )
+
+  client_link_resource_name = response.result.resource_name
+  puts "Extended an invitation from account #{manager_customer_id} to account " \
+      "#{client_customer_id} with resource name #{client_link_resource_name}."
+
+  # Find the manager_link_id of the link we just created, so we can construct
+  # the resource name for the link from the client side.
+  ga_service = client.service(:GoogleAds)
+
+  query = <<~QUERY
+    SELECT
+      customer_client_link.manager_link_id
+    FROM
+      customer_client_link
+    WHERE
+      customer_client_link.resource_name = '#{client_link_resource_name}'
+  QUERY
+
+  response = ga_service.search(manager_customer_id, query)
+  manager_link_id = response.first.customer_client_link.manager_link_id
+
+  # Accept the link using the client account.
+  client.configure do |config|
+    config.login_customer_id = client_customer_id.to_i
+  end
+
+  manager_link_service = client.service(:CustomerManagerLink)
+
+  manager_link = client.resource(:CustomerManagerLink)
+  manager_link.resource_name = client.path.customer_manager_link(
+    client_customer_id,
+    manager_customer_id,
+    manager_link_id,
+  )
+
+  mask = client.field_mask.with manager_link do
+    manager_link.status = :ACTIVE
+  end
+
+  manager_link_operation = client.operation(:CustomerManagerLink)
+  manager_link_operation['update'] = manager_link
+  manager_link_operation['update_mask'] = mask
+
+  response = manager_link_service.mutate_customer_manager_link(
+    client_customer_id,
+    [manager_link_operation],
+  )
+
+  puts "Accepted invitation with resource name #{response.results.first.resource_name}."
```

Comment:
"Client accepted invitation..."

May be a bit repetitive, but I think this example can get confusing because of the different steps executing under different roles (manager or client).